src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -163,14 +201,30 @@ export const LedgerCryptoProvider: (transport: Transport) => Promise<CryptoProvi
   ): LedgerTypes.TxOutput => {
     const changeAddressParams = getAddressParameters(changeOutputFiles, output.address, network)
     const tokenBundle = output.amount.type === TxTypes.AmountType.WITH_MULTIASSET
-      ? prepareTokenBundle(output.amount.multiasset) : null
-    const datumHashHex = output.datumHash?.toString('hex')
+      ? prepareTokenBundle(output.amount.multiasset) : undefined
+
+    const datumHashHex = prepareDatumHash(output)
+    const datum = prepareDatum(output)
+
+    const scriptHex = output.type === TxTypes.OutputType.MAP_BABBAGE
+      ? output.referenceScript?.toString('hex')
+      : undefined
 
     if (changeAddressParams && areAddressParamsAllowed(signingMode)) {
-      return prepareChangeOutput(output.amount.coin, changeAddressParams, tokenBundle, datumHashHex)
+      return prepareChangeOutput(
+        output.amount.coin,
+        changeAddressParams,
```

Comment:
It'd make sense to me if address was the first argument as it is in CDDL.

Reply:
I've completely replaced the function with `prepareDestination` in https://github.com/vacuumlabs/cardano-hw-cli/pull/127/commits/4172792c61f01ba32272c9e840aa4a773573acf1.